### ISSUE TITLE
could u provide the environment of python?

### ISSUE THREAD

**Issue by sonwe1e** [Sirui-Xu/DuMMF#4]

**Comment by Yuanhong200727** [comment]
Hey, I've ran the code successfully. Here is what my 'pip list' command list out:
`Package                        Version            Editable project location
------------------------------ ------------------ --------------------------------------------------------
absl-py                        1.0.0
aiohttp                        3.9.5
aiosignal                      1.3.1
antlr4-python3-runtime         4.9.3
anyio                          3.4.0
argon2-cffi                    21.1.0
async-timeout                  4.0.3
attrs                          21.2.0
Babel                          2.9.1
backcall                       0.2.0
bleach                         4.1.0
brotlipy                       0.7.0
cachetools                     4.2.4
certifi                        2021.5.30
cffi                           1.16.0
chamferdist                    1.0.3
charset-normalizer             3.3.2
configparser                   7.0.0
cycler                         0.11.0
debugpy                        1.5.1
decorator                      5.1.0
defusedxml                     0.7.1
e                              1.4.5
einops                         0.8.0
entrypoints                    0.3
filelock                       3.14.0
fonttools                      4.28.2
freetype-py                    2.4.0
frozenlist                     1.4.1
fsspec                         2024.3.1
fvcore                         0.1.5.post20221221
google-auth                    2.3.3
google-auth-oauthlib           0.4.6
googleapis-common-protos       1.63.0
grpcio                         1.42.0
grpcio-status                  1.33.2
human_body_prior               2.2.2.0            /home/zhengyuanhong/下载/github/DuMMF/human-body-prior/src
idna                           3.7
imageio                        2.34.1
importlib-metadata             4.8.2
importlib-resources            5.4.0
iopath                         0.1.10
ipykernel                      6.5.1
ipython                        7.29.0
ipython-genutils               0.2.0
ipywidgets                     7.6.5
jedi                           0.18.1
Jinja2                         3.0.3
json5                          0.9.6
jsonschema                     4.2.1
jupyter-client                 7.1.0
jupyter-core                   4.9.1
jupyter-server                 1.12.0
jupyterlab                     3.2.4
jupyterlab-language-pack-zh-CN 3.2.post2
jupyterlab-pygments            0.1.2
jupyterlab-server              2.8.2
jupyterlab-widgets             1.0.2
kiwisolver                     1.3.2
lazy_loader                    0.4
local-attention                1.9.1
loguru                         0.7.2
Markdown                       3.3.6
MarkupSafe                     2.0.1
matplotlib                     3.5.0
matplotlib-inline              0.1.3
mistune                        0.8.4
mpmath                         1.3.0
multidict                      6.0.5
nbclassic                      0.3.4
nbclient                       0.5.9
nbconvert                      6.3.0
nbformat                       5.1.3
nest-asyncio                   1.5.1
networkx                       3.1
ninja                          1.11.1.1
notebook                       6.4.6
numpy                          1.21.4
nvidia-cublas-cu12             12.1.3.1
nvidia-cuda-cupti-cu12         12.1.105
nvidia-cuda-nvrtc-cu12         12.1.105
nvidia-cuda-runtime-cu12       12.1.105
nvidia-cudnn-cu12              8.9.2.26
nvidia-cufft-cu12              11.0.2.54
nvidia-curand-cu12             10.3.2.106
nvidia-cusolver-cu12           11.4.5.107
nvidia-cusparse-cu12           12.1.0.106
nvidia-nccl-cu12               2.18.1
nvidia-nvjitlink-cu12          12.5.40
nvidia-nvtx-cu12               12.1.105
oauthlib                       3.1.1
omegaconf                      2.3.0
opencv-python                  4.9.0.80
packaging                      21.3
pandocfilters                  1.5.0
parso                          0.8.2
pexpect                        4.8.0
pickleshare                    0.7.5
pillow                         10.3.0
pip                            24.0
pockets                        0.9.1
portalocker                    2.8.2
prometheus-client              0.12.0
prompt-toolkit                 3.0.22
protobuf                       3.20.3
ptyprocess                     0.7.0
pyasn1                         0.4.8
pyasn1-modules                 0.2.8
pycosat                        0.6.3
pycparser                      2.22
pyDeprecate                    0.3.2
pyglet                         2.0.15
Pygments                       2.10.0
PyOpenGL                       3.1.0
pyparsing                      3.0.6
pyrender                       0.1.45
pyrsistent                     0.18.0
python-dateutil                2.8.2
pytorch-lightning              1.6.0
pytorch3d                      0.6.2
pytz                           2021.3
PyWavelets                     1.4.1
PyYAML                         6.0.1
pyzmq                          22.3.0
render                         1.0.0
render-python                  2.3.1
requests                       2.32.2
requests-oauthlib              1.3.0
rsa                            4.8
scikit-image                   0.21.0
scipy                          1.10.1
Send2Trash                     1.8.0
setuptools                     69.5.1
setuptools-scm                 6.3.2
six                            1.16.0
smplx                          0.1.28
sniffio                        1.2.0
sphinxcontrib-napoleon         0.7
supervisor                     4.2.2
sympy                          1.12
tabulate                       0.9.0
tensorboard                    2.7.0
tensorboard-data-server        0.6.1
tensorboard-plugin-wit         1.8.0
tensorboardX                   2.6.2.2
termcolor                      2.4.0
terminado                      0.12.1
testpath                       0.5.0
tifffile                       2023.7.10
tomli                          1.2.2
torch                          1.11.0+cu113
torchaudio                     0.11.0+cu113
torchgeometry                  0.1.2
torchmetrics                   0.4.1
torchvision                    0.12.0+cu113
tornado                        6.1
tqdm                           4.66.4
traitlets                      5.1.1
transforms3d                   0.3.1
trimesh                        4.3.2
triton                         2.1.0
typing_extensions              4.5.0
urllib3                        2.2.1
wcwidth                        0.2.5
webencodings                   0.5.1
websocket-client               1.2.1
Werkzeug                       2.0.2
wheel                          0.43.0
widgetsnbextension             3.5.2
yacs                           0.1.8
yarl                           1.9.4
zipp                           3.6.0`

**Comment by sonwe1e** [comment]
> Hey, I've ran the code successfully. Here is what my 'pip list' command list out: `Package Version Editable project location
> 
> absl-py 1.0.0 aiohttp 3.9.5 aiosignal 1.3.1 antlr4-python3-runtime 4.9.3 anyio 3.4.0 argon2-cffi 21.1.0 async-timeout 4.0.3 attrs 21.2.0 Babel 2.9.1 backcall 0.2.0 bleach 4.1.0 brotlipy 0.7.0 cachetools 4.2.4 certifi 2021.5.30 cffi 1.16.0 chamferdist 1.0.3 charset-normalizer 3.3.2 configparser 7.0.0 cycler 0.11.0 debugpy 1.5.1 decorator 5.1.0 defusedxml 0.7.1 e 1.4.5 einops 0.8.0 entrypoints 0.3 filelock 3.14.0 fonttools 4.28.2 freetype-py 2.4.0 frozenlist 1.4.1 fsspec 2024.3.1 fvcore 0.1.5.post20221221 google-auth 2.3.3 google-auth-oauthlib 0.4.6 googleapis-common-protos 1.63.0 grpcio 1.42.0 grpcio-status 1.33.2 human_body_prior 2.2.2.0 /home/zhengyuanhong/下载/github/DuMMF/human-body-prior/src idna 3.7 imageio 2.34.1 importlib-metadata 4.8.2 importlib-resources 5.4.0 iopath 0.1.10 ipykernel 6.5.1 ipython 7.29.0 ipython-genutils 0.2.0 ipywidgets 7.6.5 jedi 0.18.1 Jinja2 3.0.3 json5 0.9.6 jsonschema 4.2.1 jupyter-client 7.1.0 jupyter-core 4.9.1 jupyter-server 1.12.0 jupyterlab 3.2.4 jupyterlab-language-pack-zh-CN 3.2.post2 jupyterlab-pygments 0.1.2 jupyterlab-server 2.8.2 jupyterlab-widgets 1.0.2 kiwisolver 1.3.2 lazy_loader 0.4 local-attention 1.9.1 loguru 0.7.2 Markdown 3.3.6 MarkupSafe 2.0.1 matplotlib 3.5.0 matplotlib-inline 0.1.3 mistune 0.8.4 mpmath 1.3.0 multidict 6.0.5 nbclassic 0.3.4 nbclient 0.5.9 nbconvert 6.3.0 nbformat 5.1.3 nest-asyncio 1.5.1 networkx 3.1 ninja 1.11.1.1 notebook 6.4.6 numpy 1.21.4 nvidia-cublas-cu12 12.1.3.1 nvidia-cuda-cupti-cu12 12.1.105 nvidia-cuda-nvrtc-cu12 12.1.105 nvidia-cuda-runtime-cu12 12.1.105 nvidia-cudnn-cu12 8.9.2.26 nvidia-cufft-cu12 11.0.2.54 nvidia-curand-cu12 10.3.2.106 nvidia-cusolver-cu12 11.4.5.107 nvidia-cusparse-cu12 12.1.0.106 nvidia-nccl-cu12 2.18.1 nvidia-nvjitlink-cu12 12.5.40 nvidia-nvtx-cu12 12.1.105 oauthlib 3.1.1 omegaconf 2.3.0 opencv-python 4.9.0.80 packaging 21.3 pandocfilters 1.5.0 parso 0.8.2 pexpect 4.8.0 pickleshare 0.7.5 pillow 10.3.0 pip 24.0 pockets 0.9.1 portalocker 2.8.2 prometheus-client 0.12.0 prompt-toolkit 3.0.22 protobuf 3.20.3 ptyprocess 0.7.0 pyasn1 0.4.8 pyasn1-modules 0.2.8 pycosat 0.6.3 pycparser 2.22 pyDeprecate 0.3.2 pyglet 2.0.15 Pygments 2.10.0 PyOpenGL 3.1.0 pyparsing 3.0.6 pyrender 0.1.45 pyrsistent 0.18.0 python-dateutil 2.8.2 pytorch-lightning 1.6.0 pytorch3d 0.6.2 pytz 2021.3 PyWavelets 1.4.1 PyYAML 6.0.1 pyzmq 22.3.0 render 1.0.0 render-python 2.3.1 requests 2.32.2 requests-oauthlib 1.3.0 rsa 4.8 scikit-image 0.21.0 scipy 1.10.1 Send2Trash 1.8.0 setuptools 69.5.1 setuptools-scm 6.3.2 six 1.16.0 smplx 0.1.28 sniffio 1.2.0 sphinxcontrib-napoleon 0.7 supervisor 4.2.2 sympy 1.12 tabulate 0.9.0 tensorboard 2.7.0 tensorboard-data-server 0.6.1 tensorboard-plugin-wit 1.8.0 tensorboardX 2.6.2.2 termcolor 2.4.0 terminado 0.12.1 testpath 0.5.0 tifffile 2023.7.10 tomli 1.2.2 torch 1.11.0+cu113 torchaudio 0.11.0+cu113 torchgeometry 0.1.2 torchmetrics 0.4.1 torchvision 0.12.0+cu113 tornado 6.1 tqdm 4.66.4 traitlets 5.1.1 transforms3d 0.3.1 trimesh 4.3.2 triton 2.1.0 typing_extensions 4.5.0 urllib3 2.2.1 wcwidth 0.2.5 webencodings 0.5.1 websocket-client 1.2.1 Werkzeug 2.0.2 wheel 0.43.0 widgetsnbextension 3.5.2 yacs 0.1.8 yarl 1.9.4 zipp 3.6.0`

Thank you very much. Have you been able to replicate the results of the paper?

**Comment by Yuanhong200727** [comment]
> > Hey, I've ran the code successfully. Here is what my 'pip list' command list out: `Package Version Editable project location
> > absl-py 1.0.0 aiohttp 3.9.5 aiosignal 1.3.1 antlr4-python3-runtime 4.9.3 anyio 3.4.0 argon2-cffi 21.1.0 async-timeout 4.0.3 attrs 21.2.0 Babel 2.9.1 backcall 0.2.0 bleach 4.1.0 brotlipy 0.7.0 cachetools 4.2.4 certifi 2021.5.30 cffi 1.16.0 chamferdist 1.0.3 charset-normalizer 3.3.2 configparser 7.0.0 cycler 0.11.0 debugpy 1.5.1 decorator 5.1.0 defusedxml 0.7.1 e 1.4.5 einops 0.8.0 entrypoints 0.3 filelock 3.14.0 fonttools 4.28.2 freetype-py 2.4.0 frozenlist 1.4.1 fsspec 2024.3.1 fvcore 0.1.5.post20221221 google-auth 2.3.3 google-auth-oauthlib 0.4.6 googleapis-common-protos 1.63.0 grpcio 1.42.0 grpcio-status 1.33.2 human_body_prior 2.2.2.0 /home/zhengyuanhong/下载/github/DuMMF/human-body-prior/src idna 3.7 imageio 2.34.1 importlib-metadata 4.8.2 importlib-resources 5.4.0 iopath 0.1.10 ipykernel 6.5.1 ipython 7.29.0 ipython-genutils 0.2.0 ipywidgets 7.6.5 jedi 0.18.1 Jinja2 3.0.3 json5 0.9.6 jsonschema 4.2.1 jupyter-client 7.1.0 jupyter-core 4.9.1 jupyter-server 1.12.0 jupyterlab 3.2.4 jupyterlab-language-pack-zh-CN 3.2.post2 jupyterlab-pygments 0.1.2 jupyterlab-server 2.8.2 jupyterlab-widgets 1.0.2 kiwisolver 1.3.2 lazy_loader 0.4 local-attention 1.9.1 loguru 0.7.2 Markdown 3.3.6 MarkupSafe 2.0.1 matplotlib 3.5.0 matplotlib-inline 0.1.3 mistune 0.8.4 mpmath 1.3.0 multidict 6.0.5 nbclassic 0.3.4 nbclient 0.5.9 nbconvert 6.3.0 nbformat 5.1.3 nest-asyncio 1.5.1 networkx 3.1 ninja 1.11.1.1 notebook 6.4.6 numpy 1.21.4 nvidia-cublas-cu12 12.1.3.1 nvidia-cuda-cupti-cu12 12.1.105 nvidia-cuda-nvrtc-cu12 12.1.105 nvidia-cuda-runtime-cu12 12.1.105 nvidia-cudnn-cu12 8.9.2.26 nvidia-cufft-cu12 11.0.2.54 nvidia-curand-cu12 10.3.2.106 nvidia-cusolver-cu12 11.4.5.107 nvidia-cusparse-cu12 12.1.0.106 nvidia-nccl-cu12 2.18.1 nvidia-nvjitlink-cu12 12.5.40 nvidia-nvtx-cu12 12.1.105 oauthlib 3.1.1 omegaconf 2.3.0 opencv-python 4.9.0.80 packaging 21.3 pandocfilters 1.5.0 parso 0.8.2 pexpect 4.8.0 pickleshare 0.7.5 pillow 10.3.0 pip 24.0 pockets 0.9.1 portalocker 2.8.2 prometheus-client 0.12.0 prompt-toolkit 3.0.22 protobuf 3.20.3 ptyprocess 0.7.0 pyasn1 0.4.8 pyasn1-modules 0.2.8 pycosat 0.6.3 pycparser 2.22 pyDeprecate 0.3.2 pyglet 2.0.15 Pygments 2.10.0 PyOpenGL 3.1.0 pyparsing 3.0.6 pyrender 0.1.45 pyrsistent 0.18.0 python-dateutil 2.8.2 pytorch-lightning 1.6.0 pytorch3d 0.6.2 pytz 2021.3 PyWavelets 1.4.1 PyYAML 6.0.1 pyzmq 22.3.0 render 1.0.0 render-python 2.3.1 requests 2.32.2 requests-oauthlib 1.3.0 rsa 4.8 scikit-image 0.21.0 scipy 1.10.1 Send2Trash 1.8.0 setuptools 69.5.1 setuptools-scm 6.3.2 six 1.16.0 smplx 0.1.28 sniffio 1.2.0 sphinxcontrib-napoleon 0.7 supervisor 4.2.2 sympy 1.12 tabulate 0.9.0 tensorboard 2.7.0 tensorboard-data-server 0.6.1 tensorboard-plugin-wit 1.8.0 tensorboardX 2.6.2.2 termcolor 2.4.0 terminado 0.12.1 testpath 0.5.0 tifffile 2023.7.10 tomli 1.2.2 torch 1.11.0+cu113 torchaudio 0.11.0+cu113 torchgeometry 0.1.2 torchmetrics 0.4.1 torchvision 0.12.0+cu113 tornado 6.1 tqdm 4.66.4 traitlets 5.1.1 transforms3d 0.3.1 trimesh 4.3.2 triton 2.1.0 typing_extensions 4.5.0 urllib3 2.2.1 wcwidth 0.2.5 webencodings 0.5.1 websocket-client 1.2.1 Werkzeug 2.0.2 wheel 0.43.0 widgetsnbextension 3.5.2 yacs 0.1.8 yarl 1.9.4 zipp 3.6.0`
> 
> Thank you very much. Have you been able to replicate the results of the paper?

Not yet.It seems there is not evaluation code posted now. And I don't run the full training code because it takes several days.
